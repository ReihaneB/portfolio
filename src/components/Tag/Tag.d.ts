import React from 'react';

interface TagProps {
  children: React.ReactNode;
  Icon?: React.ElementType | null;
}
