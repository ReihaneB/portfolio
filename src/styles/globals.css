:root {
  /* colors */
  --white: #f5f5f7;
  --cyan: #06b6d4;
  --blue: #2756df;
  --blue-lighter: #cee5ff;
  --blue-light: #dbeafe;
  --blue-dark: #1e40af;
  --focus-blue: #93c5fd;
  --grey: #6b7280;
  --grey-light: #e5e7eb;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  border: 0;
  font: inherit;
  vertical-align: baseline;
}

html,
body {
  max-width: 100vw;
  overflow-x: hidden;
}

details,
figcaption,
figure,
footer,
hgroup,
menu,
nav {
  display: block,
}

ol,
ul {
  list-style: none;
}

blockquote,
q {
  quotes: none;
}

blockquote::before,
blockquote::after,
q::before,
q::after {
  content: '';
  content: none;
}

table {
  border-collapse: collapse;
  border-spacing: 0;
}

a {
  color: inherit;
  text-decoration: none;
}

h1 {
  font-size: 30px;
  line-height: 38px;
  font-weight: bold;
}

h2 {
  font-size: 20px;
  line-height: 24px;
  font-weight: bold;
}

h3 {
  font-size: 16px;
  line-height: 20px;
  font-weight: bold;
}

h4 {
  font-size: 14px;
  line-height: 18px;
}

h5 {
  font-size: 12px;
  line-height: 16px;
}

p {
  font-size: 16px;
  line-height: 24px;
}