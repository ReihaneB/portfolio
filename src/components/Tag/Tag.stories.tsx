import type { Meta, StoryObj } from '@storybook/react';
import { expect, userEvent, waitFor, within } from '@storybook/test';

import { lokiPlayDecorator } from '@/testUtils/loki/play-decorator';
import IconArrow from '@/styles/icons/arrows/arrow.svg';

import Tag from './Tag';

import styles from './Tag.stories.module.css';

const meta: Meta<typeof Tag> = {
  title: 'Components/Tag',
  component: Tag,
};

export default meta;
type Story = StoryObj<typeof Tag>;

const defaultArgs = {
  children: 'Click me',
};

export const Playground: Story = {
  args: {
    ...defaultArgs,
  },
};

export const WithIcon: Story = {
  args: {
    ...defaultArgs,
    Icon: IconArrow,
  },
};

export function LongText() {
  return (
    <div className={styles.container}>
      <Tag>
        This is a very long text that should be going to the next line
      </Tag>
    </div>
  );
}

LongText.storyName = 'Long text';

export function LongTextWithIcon() {
  return (
    <div className={styles.container}>
      <Tag Icon={IconArrow}>
        This is a very long text that should be going to the next line
      </Tag>
    </div>
  );
}

LongTextWithIcon.storyName = 'Long text with icon';

export const NotFocusTab: Story = {
  args: {
    ...defaultArgs,
  },
  play: lokiPlayDecorator(
    async ({ canvasElement }) => {
      const canvas = within(canvasElement);

      await userEvent.tab();

      await waitFor(() => expect(canvas.getByTestId('tag-component')).not.toHaveFocus());
    }
  ),
};
