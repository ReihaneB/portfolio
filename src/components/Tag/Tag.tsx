import { memo } from 'react';
import Image from 'next/image';

import styles from './Tag.module.css';
import type { TagProps } from './Tag.d';

function Tag({
  children,
  Icon = null,
  ...rest
}: TagProps) {
  return (
    <div
      className={styles.rootStyle}
      data-testid="tag-component"
      tabIndex={-1}
      {...rest}
    >
      {Icon && (
        <Image
          src={Icon}
          alt={`IcÃ´ne de ${children}`}
          width={12}
          height={12}
          className={styles.icon}
        />
      )}
      {children}
    </div>
  );
}

export default memo(Tag);
