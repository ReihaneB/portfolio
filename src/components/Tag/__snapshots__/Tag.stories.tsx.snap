// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Tag LongText smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-components-Tag-Tag-stories-module__container--aL11d">
    <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
         data-testid="tag-component"
         tabindex="-1"
    >
      This is a very long text that should be going to the next line
    </div>
  </div>
</main>
`;

exports[`Components/Tag LongTextWithIcon smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-components-Tag-Tag-stories-module__container--aL11d">
    <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
         data-testid="tag-component"
         tabindex="-1"
    >
      <img alt="Icône de This is a very long text that should be going to the next line"
           loading="lazy"
           width="12"
           height="12"
           decoding="async"
           data-nimg="1"
           class="src-components-Tag-Tag-module__icon--kRFOB"
           srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
           src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
           style="color: transparent;"
      >
      This is a very long text that should be going to the next line
    </div>
  </div>
</main>
`;

exports[`Components/Tag NotFocusTab play-test 1`] = `
<main class="inter-normal">
  <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
       data-testid="tag-component"
       tabindex="-1"
  >
    Tag content
  </div>
</main>
`;

exports[`Components/Tag Playground smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
       data-testid="tag-component"
       tabindex="-1"
  >
    Tag content
  </div>
</main>
`;

exports[`Components/Tag WithIcon smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
       data-testid="tag-component"
       tabindex="-1"
  >
    <img alt="Icône de Tag content"
         loading="lazy"
         width="12"
         height="12"
         decoding="async"
         data-nimg="1"
         class="src-components-Tag-Tag-module__icon--kRFOB"
         srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
         src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
         style="color: transparent;"
    >
    Tag content
  </div>
</main>
`;
